pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuda_helpers.hpp

```diff
@@ -0,0 +1,391 @@
+    std::is_same_v<T, cuDoubleComplex> || std::is_same_v<T, double2>;
+
+/**
+ * @brief Simple overloaded method to define CUDA data type.
+ *
+ * @param t
+ * @return cuDoubleComplex
+ */
+inline cuDoubleComplex getCudaType(const double &t) {
+    static_cast<void>(t);
+    return {};
+}
+/**
+ * @brief Simple overloaded method to define CUDA data type.
+ *
+ * @param t
+ * @return cuFloatComplex
+ */
+inline cuFloatComplex getCudaType(const float &t) {
+    static_cast<void>(t);
+    return {};
+}
+
+/**
+ * @brief Return the number of supported CUDA capable GPU devices.
+ *
+ * @return std::size_t
+ */
+inline int getGPUCount() {
+    int result;
+    PL_CUDA_IS_SUCCESS(cudaGetDeviceCount(&result));
+    return result;
+}
+
+/**
+ * @brief Return the current GPU device.
+ *
+ * @return int
+ */
+inline int getGPUIdx() {
+    int result;
+    PL_CUDA_IS_SUCCESS(cudaGetDevice(&result));
+    return result;
+}
+
+inline static void deviceReset() { PL_CUDA_IS_SUCCESS(cudaDeviceReset()); }
+
+/**
+ * @brief Checks to see if the given GPU supports the
+ * PennyLane-Lightning-GPU device. Minimum supported architecture is SM 7.0.
+ *
+ * @param device_number GPU device index
+ * @return bool
+ */
+static bool isCuQuantumSupported(int device_number = 0) {
+    cudaDeviceProp deviceProp;
+    cudaGetDeviceProperties(&deviceProp, device_number);
+    return deviceProp.major >= 7;
+}
+
+/**
+ * @brief Get current GPU major.minor support
+ *
+ * @param device_number
+ * @return std::pair<int,int>
+ */
+static std::pair<int, int> getGPUArch(int device_number = 0) {
+    cudaDeviceProp deviceProp;
+    cudaGetDeviceProperties(&deviceProp, device_number);
+    return std::make_pair(deviceProp.major, deviceProp.minor);
+}
+
+inline static auto pauliStringToEnum(const std::string &pauli_word)
+    -> std::vector<custatevecPauli_t> {
+    // Map string rep to Pauli enums
+    const std::unordered_map<std::string, custatevecPauli_t> pauli_map{
+        std::pair<const std::string, custatevecPauli_t>{std::string("X"),
+                                                        CUSTATEVEC_PAULI_X},
+        std::pair<const std::string, custatevecPauli_t>{std::string("Y"),
+                                                        CUSTATEVEC_PAULI_Y},
+        std::pair<const std::string, custatevecPauli_t>{std::string("Z"),
+                                                        CUSTATEVEC_PAULI_Z},
+        std::pair<const std::string, custatevecPauli_t>{std::string("I"),
+                                                        CUSTATEVEC_PAULI_I}};
+
+    static constexpr std::size_t num_char = 1;
+
+    std::vector<custatevecPauli_t> output;
+    output.reserve(pauli_word.size());
+
+    for (const auto ch : pauli_word) {
+        auto out = pauli_map.at(std::string(num_char, ch));
+        output.push_back(out);
+    }
+    return output;
+}
+
+inline static auto pauliStringToOpNames(const std::string &pauli_word)
```

Comment:
Appears unused, consider removing.

Reply:
Isn't this a requirement for the custatevec API calls for handling Pauli-words?

Reply:
In fact, this method is for the MPI backend. For the Pauli-words based `expval` calculation, we use `pauliStringToEnum` at Line 328. Let me remove it.